example_program_test.go

```diff
@@ -0,0 +1,110 @@
+// +build linux
+
+package ebpf_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/perf"
+
+	"golang.org/x/sys/unix"
+)
+
+// Example_program demonstrates how to attach an eBPF program to a tracepoint.
+// The program will be attached to the sys_enter_open syscall and print out the integer
+// 123 everytime the sycall is used.
+func Example_program() {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	events, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type: ebpf.PerfEventArray,
+		Name: "pureGo",
+	})
+	if err != nil {
+		panic(fmt.Errorf("could not create event map: %v\n", err))
+	}
+	defer events.Close()
+
+	rd, err := perf.NewReader(events, os.Getpagesize())
+	if err != nil {
+		panic(fmt.Errorf("could not create event reader: %v", err))
+	}
+	defer rd.Close()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			record, err := rd.Read()
+			if err != nil {
+				if perf.IsClosed(err) {
+					return
+				}
+				panic(fmt.Errorf("could not read from reader: %v", err))
+			}
+			fmt.Println(record)
+		}
+	}()
+
+	ins := asm.Instructions{
+		// store the integer 123 at FP[-8]
+		asm.Mov.Imm(asm.R2, int32(123)),
```

Comment:
I don't think the explicit `int32` type conversion is necessar here and also below in all cases where a constant is passed:

```suggestion
		asm.Mov.Imm(asm.R2, 123),
```